src/logics/ArithLogic.h

```diff
@@ -0,0 +1,473 @@
+    bool isNumConst (PTRef tr)  const { return isNumConst(getPterm(tr).symb()); }
+    bool isIntConst(SymRef sr) const { return isNumConst(sr) && yieldsSortInt(sr); }
+    bool isIntConst (PTRef tr) const { return isIntConst(getPterm(tr).symb()); }
+    bool isRealConst(SymRef sr) const { return isNumConst(sr) && yieldsSortReal(sr); }
+    bool isRealConst (PTRef tr) const { return isRealConst(getPterm(tr).symb()); }
+
+    bool isNonNegNumConst (PTRef tr) const { return isNumConst(tr) && isNonNegative(getNumConst(tr)); }
+
+    bool isSortInt(SRef sr) const { return sr == getSort_int(); }
+    bool isSortReal(SRef sr) const { return sr == getSort_real(); }
+    bool isSortNum(SRef sr) const { return isSortInt(sr) or isSortReal(sr); }
+
+    bool yieldsSortInt(SymRef sr) const { return isSortInt(sym_store[sr].rsort()); }
+    bool yieldsSortInt(PTRef tr)  const { return yieldsSortInt(getPterm(tr).symb()); }
+    bool yieldsSortReal(SymRef sr) const { return isSortReal(sym_store[sr].rsort()); }
+    bool yieldsSortReal(PTRef tr)  const { return yieldsSortReal(getPterm(tr).symb()); }
+    bool yieldsSortNum(SymRef sr) const { return yieldsSortInt(sr) or yieldsSortReal(sr); }
+    bool yieldsSortNum(PTRef tr)  const { return yieldsSortInt(tr) or yieldsSortReal(tr); }
+
+    const FastRational & getNumConst(PTRef tr) const;
+
+    bool           isUFEquality(PTRef tr) const override { return !isNumEq(tr) && Logic::isUFEquality(tr); }
+    bool           isTheoryEquality(PTRef tr) const override { return isNumEq(tr); }
+    bool           isAtom(PTRef tr) const override { return isNumEq(tr) || isLt(tr) || isGt(tr) || isLeq(tr) || isGeq(tr) || Logic::isAtom(tr); }
+    const char*    getDefaultValue(PTRef tr) const override;
+    PTRef          getDefaultValuePTRef(SRef sref) const override;
+
+
+    SymRef get_sym_Int_TIMES () const { return sym_Int_TIMES; }
+    SymRef get_sym_Real_TIMES () const { return sym_Real_TIMES; }
+    SymRef get_sym_Int_DIV () const { return sym_Int_DIV; }
+    SymRef get_sym_Int_MOD () const { return sym_Int_MOD; }
+    SymRef get_sym_Real_DIV () const { return sym_Real_DIV; }
+    SymRef get_sym_Int_MINUS () const { return sym_Int_MINUS; }
+    SymRef get_sym_Real_MINUS () const { return sym_Real_MINUS; }
+    SymRef get_sym_Real_PLUS () const {return sym_Real_PLUS; }
+    SymRef get_sym_Int_PLUS () const {return sym_Int_PLUS; }
+    SymRef get_sym_Int_NEG () const { return sym_Int_NEG; }
+    SymRef get_sym_Int_LEQ() const { return sym_Int_LEQ; }
+    SymRef get_sym_Real_LEQ() const { return sym_Real_LEQ; }
+    SymRef getLeqForSort(SRef sr) const;
+    SymRef get_sym_Int_GEQ() const { return sym_Int_GEQ; }
+    SymRef get_sym_Real_GEQ() const { return sym_Real_GEQ; }
+    SymRef get_sym_Int_LT() const { return sym_Int_LT; }
+    SymRef get_sym_Real_LT() const { return sym_Real_LT; }
+    SymRef get_sym_Int_GT() const { return sym_Int_GT; }
+    SymRef get_sym_Real_GT() const { return sym_Real_GT; }
```

Comment:
Do we actually need these symbols? We are representing strict inequalities with negations of non-strict inequalities, no? We do not actually use these, right?

Reply:
No, but I would still keep them there for the future plans. In particular I'm thinking of filling the gaps in our proof production, when we'd need to show the related transformations.

Reply:
OK